### PR TITLE
Prevent erroneous validation failure for primitive type conversion

### PR DESCRIPTION
**Pull request by geoand**
Relates to: https://github.com/quarkusio/quarkus/issues/29918

### CODE REVIEW

**Comment by geoand** [comment]
Actually this needs a little more polish

File: implementation/src/main/java/io/smallrye/config/ConfigMappingInterface.java [hunk 1]
```diff
@@ -27,6 +28,22 @@
...
+        map.put(char.class, Character.class);
+        map.put(byte.class, Byte.class);
+        map.put(short.class, Short.class);
+        map.put(int.class, Integer.class);
+        map.put(long.class, Long.class);
+        map.put(float.class, Float.class);
+        map.put(double.class, Double.class);
+
+        PRIMITIVES_TO_WRAPPERS = Collections.unmodifiableMap(map);
+    }
```
Comment (geoand) [inline implementation/src/main/java/io/smallrye/config/ConfigMappingInterface.java:45]: Is there perhaps a better place to put this?

**Comment by geoand** [comment]
No problem!

I'll check out your suggestion tomorrow

File: implementation/src/main/java/io/smallrye/config/ConfigMappingInterface.java (hunk 1 above)
Comment (geoand) [reply implementation/src/main/java/io/smallrye/config/ConfigMappingInterface.java]: PR updated. WDYT?

Comment (radcortez) [reply implementation/src/main/java/io/smallrye/config/ConfigMappingInterface.java]: Looks great! Thanks!

**Review by radcortez — APPROVED** [review]

**Comment by geoand** [comment]
👌

**Pull request merged**